### PR TITLE
Item form modal

### PR DESCRIPTION
**Pull request by allishultes**
**Is your Pull Request request related to another [issue](https://github.com/bbc/digital-paper-edit-storybook/issues) in this repository ?**      

Issue #7 
Issue #10 

**Describe what the PR does**    

Adds the ItemForm and FormModal components to the Storybook. Both have been refactored using React Hooks. 

The ItemForm has one story: `Edit Project`. It:

- [ ] Takes a title (string) and description (string) as inputs
- [ ] Validates the title and does not allow a project to be saved without one
- [ ] Logs an 'Item Saved' action when a valid project is saved

The ItemFormModal component has two stories: `Edit Project` and `New Project`. The `Edit Project` story includes an ItemForm component that is passed a title, description and id from the FormModal component. The `New Project` story does not pas ItemForm an id, title or description. In both cases, the modal:

- [ ] Has a 'close' button which closes the modal;
- [ ] Displays the ItemForm component

The PR also changes the folder structure for the List and Search Bar components for kicks and giggles. 

**State whether the PR is ready for review or whether it needs extra work**    

Almost ready for review (where does the new project id come from?).

**Additional context**

### CODE REVIEW

File: packages/components/FormModal/ItemForm/index.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      </Modal.Footer>
+    </Form>
+  );
+};
+
+ItemForm.propTypes = {
+  id: PropTypes.number.isRequired,
+  title: PropTypes.string,
+  description: PropTypes.string,
+  isNewItemModalShow: PropTypes.bool.isRequired,
```
Comment (emettely) [inline packages/components/FormModal/ItemForm/index.js]: Also maybe rename, it's not clear what it is.

Comment (emettely) [reply packages/components/FormModal/ItemForm/index.js]: something like, `showModal` as seen somewhere else :)

File: packages/components/FormModal/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'react-bootstrap/Modal';
+import ItemForm from './ItemForm/index.js';
+
+const ItemFormModal = (props) => {
+
+  const [ showModal, toggleShowModal ] = useState(props.isNewItemModalShow);
```
Comment (emettely) [inline packages/components/FormModal/index.js]:
```suggestion
  const [ showModal, setShowModal ] = useState(props.showModal);
```

File: packages/components/FormModal/ItemForm/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import PropTypes from 'prop-types';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+
+const ItemForm = (props) => {
+
+  const [ description, setDescription ] = useState(props.description);
+
+  const [ isValidated, setFormValidation ] = useState(false);
```
[the remark this answers is not in the document]
Comment (allishultes) [reply packages/components/FormModal/ItemForm/index.js]: Ok cool - I'll be sure to check out the docs next time I add hooks; I've not been following them so much as working from example.

File: packages/components/FormModal/ItemForm/index.js [hunk 4]
```diff
@@ -8,28 +8,25 @@ const ItemForm = (props) => {
...
       const editedProject = {
         title: title,
         description: description,
         id: props.id
       };
 
       props.handleSaveForm(editedProject);
-    }
+    } else if (!isValidated) {
+      setIsValidated(true);
```
Comment (emettely) [inline packages/components/FormModal/ItemForm/index.js]: `setIsValidated(true);` needs to happen for both cases - see line 20 + 24.

File: packages/components/FormModal/index.js [hunk 5]
```diff
@@ -5,7 +5,9 @@ import ItemForm from './ItemForm/index.js';
 
 const ItemFormModal = (props) => {
 
-  const [ showModal, toggleShowModal ] = useState(props.isNewItemModalShow);
+  console.log('modal props', props);
```
Comment (emettely) [inline packages/components/FormModal/index.js]:
```suggestion
```

**Pull request merged**